src/providers/types.ts

```diff
@@ -153,3 +154,22 @@ export interface ImageGenerateResponse {
   data: object[];
   provider: string;
 }
+
+/**
+ * The structure of a completion response for the 'fimComplete' function.
+ * @interface
+ */
+export interface FimCompletionResponse extends CResponse {
+  choices: FimChoice[];
+}
+
+/**
+ * The structure of a choice in a chat completion response.
+ * @interface
+ */
+export interface FimChoice {
+  index: number;
+  message: Message;
+  finish_reason: string;
```

Comment:
Recommend making this an enum
```
enum FIM_FINISH REASON {
  stop = 'stop',
  ...
}
```

Reply:
on the mistral AI documentation it  it is written that it can be it can be array or string. what should i do. 
![image](https://github.com/user-attachments/assets/6ed88a90-66e2-4eae-a30d-96db91f16050)

i have following two options:
1) 
```
enum FIM_FINISH_REASON {
  stop = 'stop',
}
export interface FimChoice {
  index: number;
  message: Message;
  finish_reason: FIM_FINISH_REASON;
  logprobs?: object | null;
}
```

2)

```
export interface FimChoice {
  index: number;
  message: Message;
  finish_reason: string | string[];  // Union type for string or array of strings
  logprobs?: object | null;
}

```

Reply:
done